WordPress/src/main/java/org/wordpress/android/models/ReaderTag.java

```diff
@@ -180,11 +180,12 @@ public String getLabel() {
     }
 
     @Override
-    public boolean equals(Object tag){
-        if (tag == null) return false;
-
-        if (!ReaderTag.class.isAssignableFrom(tag.getClass())) return false;
-
-        return getLabel().equals(((ReaderTag) tag).getLabel());
+    public boolean equals(Object object){
+        if (object instanceof ReaderTag) {
+            ReaderTag tag = (ReaderTag) object;
+            return (tag.tagType == this.tagType && tag.getLabel().equals(this.getLabel()));
```

Comment:
Reading this line only there's a possibility that `tag.getLabel()` returns null - we know by looking into `getLabel()` implementation that it's impossible for it to return null, but it's not evident on this line. What do you think, should we add a comment stating that we _expect_ `getLabel()` will return a non null value?